src/main/java/io/logz/jmx2graphite/JavaAgentClient.java

```diff
@@ -0,0 +1,183 @@
+import javax.management.openmbean.TabularData;
+import java.lang.management.ManagementFactory;
+import java.time.LocalDate;
+import java.time.ZoneId;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Created by roiravhon on 6/6/16.
+ */
+public class JavaAgentClient extends MBeanClient {
+
+    private static final Logger logger = LoggerFactory.getLogger(JavaAgentClient.class);
+
+    MBeanServer server;
+    ObjectMapper objectMapper;
+
+    public JavaAgentClient() {
+
+        server = ManagementFactory.getPlatformMBeanServer();
+        objectMapper = new ObjectMapper();
+        objectMapper.setVisibility(PropertyAccessor.ALL, JsonAutoDetect.Visibility.NONE);
+        objectMapper.setVisibility(PropertyAccessor.FIELD, JsonAutoDetect.Visibility.ANY);
+    }
+
+    @Override
+    public List<MetricBean> getBeans() throws MBeanClientPollingFailure {
+
+        try {
+            List<MetricBean> metricBeans = Lists.newArrayList();
+            Set<ObjectInstance> instances = server.queryMBeans(null, null);
+
+            for (ObjectInstance instance : instances) {
+                MBeanInfo mBeanInfo = server.getMBeanInfo(instance.getObjectName());
+                List<String> attributes = Lists.newArrayList();
+
+                for (MBeanAttributeInfo attribute : mBeanInfo.getAttributes()) {
+                    attributes.add(attribute.getName());
+                }
+
+                // Dont change to getCanonicalName(), we need it to preserve the order so we can have a valuable metrics tree
+                metricBeans.add(new MetricBean(instance.getObjectName().getDomain() + ":" + instance.getObjectName().getKeyPropertyListString(), attributes));
+            }
+            return metricBeans;
+
+        } catch (IntrospectionException | ReflectionException | InstanceNotFoundException e) {
+            throw new MBeanClientPollingFailure(e.getMessage(), e);
+        }
+    }
+
+    @Override
+    public List<MetricValue> getMetrics(List<MetricBean> beans) throws MBeanClientPollingFailure{
+
+        List<MetricValue> metricValues = Lists.newArrayList();
+
+        for (MetricBean metricBean : beans) {
+            try {
+                AttributeList attributeList = server.getAttributes(new ObjectName(metricBean.getName()),
+                                                                   metricBean.getAttributes().toArray(new String[0]));
+
+                Map<String, Object> attrValues = new HashMap<>(attributeList.size());
+                attributeList.asList().forEach((attr) ->
+                        attrValues.put(attr.getName(), attr.getValue()));
+
+                Map<String, Number> metricToValue = flatten(attrValues);
+
+                for (String attrMetricName : metricToValue.keySet()) {
+                    try {
+
+                        long metricTime = LocalDate.now().atStartOfDay(ZoneId.of("UTC")).toEpochSecond();
```

Comment:
Quoting the java doc of asStartOfDay:
`    /**
     * Combines this date with the time of midnight to create a {@code LocalDateTime}
     * at the start of this date.
     * <p>
     * This returns a {@code LocalDateTime} formed from this date at the time of
     * midnight, 00:00, at the start of this date.
     *
     * @return the local date-time of midnight at the start of this date, not null
     */
    public LocalDateTime atStartOfDay() {
        return LocalDateTime.of(this, LocalTime.MIDNIGHT);
    }
`
Test it and see you are truncating the hour, min and second